indico/modules/events/registration/controllers/management/regforms.py

```diff
@@ -292,3 +302,48 @@ def _process(self):
             update_object_principals(self.event, form.managers.data, permission='registration')
             return jsonify_data(flash=False)
         return jsonify_form(form)
+
+
+class RHAddRegistrationTag(RHManageRegFormsBase):
+    """Add a new registration tag."""
+
+    def _process(self):
+        form = RegistrationTagForm()
+        if form.validate_on_submit():
+            tag = RegistrationTag(event_id=self.event.id)
```

Comment:
similar to my comment above; don't use IDs. you can just do `event=self.event` or `self.event.registration_tags.append(tag)`. in either case, `db.session.add()` won't be needed